File: src/main/java/de/nordgedanken/auto_hotkey/AHKFileType.java
package de.nordgedanken.auto_hotkey;

import com.intellij.openapi.fileTypes.LanguageFileType;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

public class AHKFileType extends LanguageFileType {
    public static final AHKFileType INSTANCE = new AHKFileType();

    private AHKFileType() {
        super(AHKLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public String getName() {
        return "AHK Script";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "AutoHotKey script file";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return "ahk";
    }

    @Nullable
    @Override
    public Icon getIcon() {
        return AHKIcons.FILE;
    }

}

File: src/main/java/de/nordgedanken/auto_hotkey/AHKFileType.kt
package de.nordgedanken.auto_hotkey

import com.intellij.openapi.fileTypes.LanguageFileType
import com.intellij.openapi.vfs.VirtualFile
import javax.swing.Icon

object AHKFileType : LanguageFileType(AHKLanguage) {

    override fun getName(): String = "AHK script"

    override fun getIcon(): Icon = AHKIcons.FILE

    override fun getDefaultExtension(): String = "ahk"

    override fun getCharset(file: VirtualFile, content: ByteArray): String = "UTF-8"

    override fun getDescription(): String = "AutoHotKey script file"
}
